netket/experimental/operator/_fermion_operator_2nd.py

```diff
@@ -106,15 +110,17 @@ def __init__(
         super().__init__(hilbert)
 
         # bring terms, weights into consistent form, autopromote dtypes if necessary
-        _operators, _constant, dtype = _canonicalize_input(
+        _terms, _weights, _constant, dtype = _canonicalize_input(
             terms, weights, constant, dtype
         )
-        _verify_input(hilbert, _operators, raise_error=True)
+        _verify_input(hilbert, _terms, _weights)
         self._dtype = dtype
 
         # we keep the input, in order to be able to add terms later
-        self._operators = _operators
+        self._terms = _terms
+        self._weights = _weights
         self._constant = _constant
+        self.reduce()
```

Comment:
actually I would not even call this here. 
It's called all the time from the fermion constructors so this is wasting time...